projects/plugins/jetpack/extensions/shared/external-media/media-service/index.ts

```diff
@@ -279,11 +294,36 @@ export const authenticateMediaSource = ( source: MediaSource, isAuthenticated: b
  * @param {PickerSession} session
  */
 export const setGooglePhotosPickerSession = ( session: PickerSession ) => {
+	setGooglePhotosPickeCachedSessionId( session?.id || null );
 	dispatch( mediaStore ).mediaPhotosPickerSessionSet( session );
 };
 
 /**
  * Get Google Photos Picker session
  * @return {PickerSession} Media URL.
  */
-export const getGooglePhotosPickerSession = () => select( mediaStore ).mediaPhotosPickerSession();
+export const getGooglePhotosPickerSession = () => {
+	return select( mediaStore ).mediaPhotosPickerSession();
+};
+
+/**
+ * Set Google Photos Picker session id to cookies
+ * @param {string|null} sessionId - Session id
+ */
+export const setGooglePhotosPickeCachedSessionId = ( sessionId: string | null ) => {
+	wpCookies.set(
+		GOOGLE_PHOTOS_PICKER_SESSION,
+		sessionId,
+		604800, // 7 days
+		'/',
+		`.${ window.location.hostname }`
```

Comment:
This now works for the Simple sites, thanks for handling this! But I'm wondering how does this works for Atomic and Jetpack site since this will be their own domain. Or do we need to do extra handling here?

Reply:
As far as I know. We are going away from the calypso experience if it's an atomic site.
For example, if you select the atomic site from the `/sites` page, you will be redirected to the wp-admin experience, so we don't need to care about it. Isn't it?

Reply:
General speaking I'd say yes, but I'm not sure about the media scenario. For now, if I click media library on an Atomic site, some of my sites will go to `/wp-admin` version, but some remains the default WordPress.com experience. If this is the case, it sounds to me that for Atomic users they won't be able to add images from the external media in the media library in the future, but this discussion is probably out of scope of the PR.

I think for now it's fine for us to leave it like this, we can tackle it again once we have more clear picture on how we handle media in Atomic in the future.